### PR TITLE
LRQA-66784 | master

### PR DESCRIPTION
**Pull request by weller0246**
https://issues.liferay.com/browse/LRQA-66784

### CODE REVIEW

**Comment by weller0246** [comment]
Please review my changes @gaborlovasdotcom, @JoyceWang08211

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Review by gaborlovasdotcom — COMMENTED** [review]
LGTM

**Comment by gaborlovasdotcom** [comment]
@JoyceWang08211 Could you please review it next week?
@weller0246 Could you please make an example test case where you utilize all macro's you added?

**Comment by weller0246** [comment]
ci:test:staging:nocompile

**Comment by weller0246** [comment]
ci:test:staging:nocompile

**Comment by weller0246** [comment]
ci:test:staging:nocompile

**Comment by JoyceWang08211** [comment]
Suggestion:
1. It's better to use the same structure as other JSON files, aka JSONSiteTemplate.macro, JSONSiteTemplateSetter.macro and JSONSiteTemplateAPI.macro. It's more convenient to maintain.
a. JSONSiteTemplateGetter.getSiteTemplateIdByName and JSONSiteTemplateAPI._getSiteTemplateNameById ----> Combine into one macro.
b. JSONSiteTemplateGetter.getSiteTemplateUuidByName and JSONSiteTemplateAPI._getSiteTemplateUuidByName ----> Combine into one macro.

2. Intricate nested calls in JSONSiteTemplateAPI._getSiteTemplateNameById and JSONSiteTemplateAPI._getSiteTemplateUuidByName.
Replace get-layout-set-prototype with get-layout-set-prototypes, then no need to pass in **-d layoutSetPrototypeId=${templateId}**.
e.g.
```
var curl = '''
			${portalURL}/api/jsonws/layoutsetprototype/get-layout-set-prototypes \
				-u test@liferay.com:test \
				-d companyId=${companyId}
		''';
```
Then use **nameCurrentValue** to filter templateId(aka, layoutSetPrototypeId) or siteTemplateUuid(aka uuid) from the raw curl results. Take a reference from [here](https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutAPI.macro#L145)
e.g.
`var layoutSetPrototypeId = JSONCurlUtil.post("${curl}", "$.[?(@['nameCurrentValue'] == '${siteTemplateName}')]['layoutSetPrototypeId']");`

In this way, we can reduce unnecessary nested calls and more readable. 

3. Useless macro JSONSiteTemplateAPI._getSiteTemplateById, it isn't called by other macro. 

4. All sitetemplate testcases are running on 8080, we don't have the specific needs for portalURL. I think **var portalURL = JSONCompany.getPortalURL();** is good enough, no need to add a condition for portalURL.

5. Need to expand JSONSiteTemplate.addLayoutToSiteTemplate for adding child page. We have a [testcase](https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/tests/enduser/sitetemplates/usecase/SitetemplatesUsecase.testcase#L114) need to add child page in site template. 

6. Some SF issues, e.g. Variables.assertDefined(parameterList = "**${siteTemplateName},${groupName}**");
It's better to be like Variables.assertDefined(parameterList = "**${groupName},${siteTemplateName}**"); and so on.

**Comment by weller0246** [comment]
I resolved the points of your review, please review my changes. @gaborlovasdotcom  @JoyceWang08211

File: portal-web/test/functional/com/liferay/portalweb/macros/json/site/JSONSiteTemplateAPI.macro [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+			${portalURL}/api/jsonws/layoutset/update-layout-set-prototype-link-enabled \
+                           -u test@liferay.com:test \
+                           -d groupId=${groupId}  \
+                           -d privateLayout=${privateLayout} \
+                           -d layoutSetPrototypeLinkEnabled=true \
+                           -d layoutSetPrototypeUuid=${templateUuid}
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/macros/json/site/JSONSiteTemplateAPI.macro:32]: layoutSetPrototypeUuid is expected as incoming parameter name

**Review by gaborlovasdotcom — CHANGES_REQUESTED** [review]
Looks good to me, only one parameter name seems off.

File: portal-web/test/functional/com/liferay/portalweb/macros/json/site/JSONSiteTemplateAPI.macro (hunk 1 above)
Comment (weller0246) [reply portal-web/test/functional/com/liferay/portalweb/macros/json/site/JSONSiteTemplateAPI.macro]: Nice catch, thanks! I found actually a few unchanged parameter name so it was a good point! Resolved

**Comment by JoyceWang08211** [comment]
Very concise logic of the refactor, good job @weller0246

**Review by gaborlovasdotcom — APPROVED** [review]

**Comment by weller0246** [comment]
Continued here: https://github.com/gaborlovasdotcom/liferay-portal/pull/299

**Pull request closed without merging**